src/browser/ui/ReactDOMComponent.js

```diff
@@ -235,7 +235,7 @@ ReactDOMComponent.Mixin = {
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
         var markup =
-          DOMPropertyOperations.createMarkupForProperty(propKey, propValue);
+          DOMPropertyOperations.createMarkupForProperty(this._tag, propKey, propValue);
```

Comment:
This is leaking unnecessarily much information to DOMPropertyOperations. All the tag name logic is already in this file.

I'd rather see the switching done in this file where we already have a regexp for tag names. We might as well use that first check to cache if we're a custom element or not. That way you don't have to recheck for every attribute and every update.